src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts

```diff
@@ -0,0 +1,607 @@
+  public selectedFilterPlugin;
+  changedFilterConfig: any;
+  public deletedFilterPipeline = [];
+  public isFilterDeleted = false;
+  confirmationDialogData = {};
+  public isAddFilterWizard;
+
+  @ViewChild('filterConfigComponent') filterConfigComponent: ConfigurationGroupComponent;
+  @ViewChild(FilterAlertComponent) filterAlert: FilterAlertComponent;
+
+  constructor(
+    private cdRef: ChangeDetectorRef,
+    private assetService: AssetsService,
+    private route: ActivatedRoute,
+    private controlPipelinesService: ControlPipelinesService,
+    private alertService: AlertService,
+    private ngProgress: ProgressBarService,
+    private dialogService: DialogService,
+    public sharedService: SharedService,
+    private schedulesService: SchedulesService,
+    private controlService: ControlDispatcherService,
+    public notificationService: NotificationsService,
+    private filterService: FilterService,
+    private docService: DocService,
+    private configurationControlService: ConfigurationControlService,
+    private router: Router) { }
+
+  ngOnInit(): void {
+    this.getSourceDestTypes('source');
+    this.getSourceDestTypes('destination');
+    this.route.params.subscribe(params => {
+      this.pipelineID = params['id'];
+      if (this.pipelineID) {
+        this.editMode = true;
+        this.getControlPipeline();
+      } else {
+        this.selectedExecution = 'Shared';
+        this.selectedSourceType = {cpsid: 1, name: "Any"};
+        this.selectedDestinationType = {cpdid: 4, name: "Broadcast"};
+      }
+    });
+    this.getAllPipelines(); 
+  }
+
+  ngAfterContentChecked(): void {
+    this.cdRef.detectChanges();
+  }
+
+  refresh() {
+    this.getControlPipeline();
+  }
+
+  onDrop(event: CdkDragDrop<string[]>) {
+    if (event.previousIndex === event.currentIndex) {
+      return;
+    }
+    moveItemInArray(this.filterPipeline, event.previousIndex, event.currentIndex);
+    this.isFilterOrderChanged = true;
+    this.pipelineForm.form.markAsDirty();
+  }
+
+  // savePipelineChanges() {
+  //   if (this.isFilterDeleted) {
+  //     this.deleteFilter();
+  //   }
+  // }
+
+  openAddFilterModal(isClicked, nameValue) {
+    if (!nameValue || nameValue === ''){
+      return;
+    }
+    if (this.isFilterOrderChanged || this.isFilterDeleted) {
+      this.showConfirmationDialog();
+      return;
+    }
+    this.isAddFilterWizard = isClicked;
+    this.isFilterOrderChanged = false;
+    this.isFilterDeleted = false;
+    this.deletedFilterPipeline = [];
+  }
+
+  /**
+  * Open confirmation modal
+  */
+   showConfirmationDialog() {
+    this.confirmationDialogData = {
+      id: '',
+      name: '',
+      message: 'Do you want to discard unsaved changes?',
+      key: 'unsavedConfirmation'
+    };
+    this.filterAlert.toggleModal(true);
+  }
+
+  onNotify(data) {
+    const filterData = data ? data.filters : [];
+    let filterNameList = [];
+    // Append recently added filter to existing filter pipeline
```

Comment:
To avoid confusion with parent object Control Pipeline, let call here it filters list